### ISSUE TITLE
Training object_detection model doesn't evaluate properly

### ISSUE THREAD

**Issue by CharlesAverill** [tensorflow/models#7133]
### System information
- **What is the top-level directory of the model you are using**: ssd_mobilenet_v1_coco_2018_01_28
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: 1.14.0
- **Bazel version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: (I'm running this from Google Colab with GPU)
- **GPU model and memory**: (I'm running this from Google Colab with GPU)
- **Exact command to reproduce**: 

> python models/research/object_detection/model_main.py \
    --pipeline_config_path="./training/pipeline.config" \
    --model_dir='pretrained_model/saved_model' \
    --sample_1_of_n_eval_samples 1 \
    --alsologtostderr \
    --num_train_steps={newmodelno} \

### Describe the problem
Whenever I train my custom object_detection model, the evaluation block is always incorrect, here's what is output almost 100% of the time:

> I0701 19:24:52.836364 139760421549824 coco_tools.py:109] Loading and preparing annotation results...
I0701 19:24:52.840913 139760421549824 coco_tools.py:131] DONE (t=0.00s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.04s).
Accumulating evaluation results...
DONE (t=0.01s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = -1.000

The precision is always -1.000, every single time, which has to be false because I can see the output and it does indeed display detection rectangles with positive confidence values (it's not accurate yet but I figure that's just because I've only trained it for about 20,000 steps). Back when I was training offline it did the same thing. When I started using a script to automatically generate the pipeline.config file instead of me personally editing it, the problem persisted. I have no idea why this happens and can't find any info about it. I believe it's also related to this error I get when converting the checkpoints to a frozen model:

> ==================Model Analysis Report======================
Incomplete shape.

Can't find anything about this either. I'm really confused because I followed tutorials word for word. The only thing I can imagine is wrong is my custom dataset, but I just made it with LabelImg in the PascalVOC format like the tutorials said.

Maybe this is also related to the fact that my loss value starts around .34 and doesn't ever significantly change?

I appreciate your help

**Comment by CharlesAverill** [comment]
I just started over with the stock tensorflow-object-detection-training-colab.ipynb file and everything works great now.

**Comment by tommy3266** [comment]
I have fix the same issue,it is because the content of  my label_map.pbtxt is wrong!!
